src/usePlaidLink.ts

```diff
@@ -57,7 +57,7 @@ export const usePlaidLink = (options: PlaidLinkOptions) => {
 
     // destroy the Plaid iframe factory
     return () => next.exit({ force: true }, () => next.destroy());
-  }, [loading, error, options.token]);
+  }, [loading, error, options.token, options.product]);
```

Comment:
As the array is shallow compared:

```suggestion
    }, [loading, error, options.token, (options.product || []).slice().sort().join(',')]);
```